server/aesdsocket.c

```diff
@@ -0,0 +1,390 @@
+        goto exit_segment;
+	}    
+ 
+
+
+
+	while(1)
+	{
+
+        client_fd = accept(server_fd, (struct sockaddr*)&client_addr, &addrlen);
+        
+        if (client_fd < 0)
+        {
+            perror("While accepting client");
+            error = true;
+            goto exit_segment;
+        } 
+
+        // Block the signals SIGINT and SIGTERM here
+        if (sigprocmask(SIG_BLOCK, &intmask, NULL) < 0)
+        {
+            error = true;
+            goto exit_segment;
+        }
+        
+        char s[INET6_ADDRSTRLEN];  
+
+        inet_ntop(AF_INET, get_in_addr((struct sockaddr *)&client_addr), s, sizeof s);
+        syslog(LOG_DEBUG,"Accepted connection from %s", s);   
+
+        // TODO : handle failure ret values
+
+		int read_size = 0;
+		int ret = 0;
+        bool terminate = false;
+        char *dyn_buffer1 = NULL;
+        char *dyn_buffer2 = NULL;
+        long size = BUFFER_SIZE;
+        long cur_size = 0;
+        int multiplier = 2;
+        long curr_pos = 0;
+
+        dyn_buffer1 = (char *)malloc(sizeof(char) * BUFFER_SIZE);
+
+        if (dyn_buffer1 == NULL)
+        {
+            syslog(LOG_ERR, "(dyn_buffer1) malloc returned NULL");
+            error = true;
+            goto exit_segment;
+        }
+
+
+        do 
+        {
+            read_size = recv(client_fd, buffer, sizeof(buffer), 0);
+
+            if (read_size < 0)
+            {
+                perror("While receiving socket data");
+                error = true;
+                goto exit_segment;
+
+            }
+
+            if (!read_size || (strchr(buffer, '\n') != NULL)) terminate = true;
+
+            // dyn_buffer1 full, must resize
+            if ((size - cur_size) < read_size)
+            {
+                do
+                {                                    
+                    size = BUFFER_SIZE * multiplier;
+                    multiplier++;
+
+                } while ((size - cur_size) < read_size);
+
+                dyn_buffer1 = (char *)realloc(dyn_buffer1, sizeof(char) * size);
+                
+                if (dyn_buffer1 == NULL)
+                {
+                    syslog(LOG_ERR, "(dyn_buffer1) realloc returned NULL");
+                    error = true;
+                    goto exit_segment;
+                }
+                
+            }
+
+            memcpy(&dyn_buffer1[cur_size], buffer, read_size);
+            cur_size += read_size;
+                
+
+  
+
+        } while (terminate == false);
+
+        // write to file
+        ret = write(fd, dyn_buffer1, cur_size);
+        if (ret < 0)
+        {
+            perror("While writing to file");
+            error = true;
+            goto exit_segment;
+            
+        }
+
+        lin_sz_table[line_sz_index++] = cur_size;
+     
+        // send packet by packet i.e line by line
+        
+        curr_pos = lseek(fd, 0, SEEK_CUR);
+        lseek(fd, 0, SEEK_SET);
+
+        for (int i = 0; i < line_sz_index; i++)
```

Comment:
@SundarKrishnakumar this works to manage RAM heap but is more complicated than just reading out the file in chunks and looking for newline to split up packets.  It also has the problem mentioned above and other limitations related to threading, for instance if multiple threads are modifying the file between the time `line_sz_table` is modified and the time the file is read.